web/modules/custom/joinup_federation/src/Plugin/pipeline/Step/UserSelectionFilter.php

```diff
@@ -169,34 +169,71 @@ public function buildConfigurationForm(array $form, FormStateInterface $form_sta
       '#markup' => $this->t("Check the solutions that you wish to import. Unselected solutions will be saved to 'Blacklisted Solutions' section on the next import attempt. You can always import a blacklisted solution by selecting it."),
     ];
 
-    $options = $default_value = [];
+    $valid_options = $invalid_options = $default_value = [];
     foreach ($entities_per_category as $category_id => $entities) {
       foreach ($entities as $id => $label) {
-        $options[$id] = [
+        $data = [
           'solution' => $label,
           'info' => $this->getInfo($id, $category_id),
           '#attributes' => ['data-drupal-federation-category' => $category_id],
         ];
-        $default_value[$id] = $category_id !== 'blacklisted' ? $id : NULL;
+
+        if ($category_id === 'invalid_collection') {
+          $invalid_options[$id] = $data;
+          unset($entities[$id]);
+        }
+        else {
+          $valid_options[$id] = $data;
+          $default_value[$id] = $category_id !== 'blacklisted' ? $id : NULL;
+        }
       }
-      $options += $entities;
+      $valid_options += $entities;
+    }
+
+    if (isset($entities_per_category['invalid_collection'])) {
+      $invalid_entities = $entities_per_category['invalid_collection'];
+      unset($entities_per_category['invalid_collection']);
     }
 
     $form['user_selection'] = [
       '#type' => 'tableselect',
-      '#options' => $options,
+      '#options' => $valid_options,
       '#header' => [
         'solution' => $this->t('Solution'),
         'info' => $this->t('Info'),
       ],
-      '#empty' => $this->t('This import contains no incoming entities. Nothing will be imported.'),
+      '#empty' => $this->t('This import contains no entities that can be federated. Nothing will be imported.'),
       '#default_value' => $default_value,
       '#after_build' => [
         // We'll append a new pre-render callback.
         [static::class, 'alterFormPreRender'],
       ],
     ];
 
+    if (!empty($invalid_entities)) {
+      $info = self::getCategoryMetadata('invalid_collection');
+      $form['invalid_collection'] = [
+        [
+          '#prefix' => '<h4>',
+          '#suffix' => '</h4>',
+          '#markup' => $info['label'],
+        ],
+        [
+          '#prefix' => '<p>',
+          '#suffix' => '</p>',
+          '#markup' => $info['description'],
+        ],
+        [
+          '#type' => 'table',
```

Comment:
This remains to be discussed, however, I do enjoy the idea to be honest, to have a complete supported tableselect UI and since you already did some work on the upstream patch, I like the idea of how it makes our code more compact (because I had to write extra render arrays to support the new list).